tutorials/developing-services-with-k8s.md

```diff
@@ -0,0 +1,196 @@
+
+```
+% kubectl create -f redis-master-deployment.yaml
+% kubectl create -f redis-master-service.yaml
+% kubectl create -f redis-slave-deployment.yaml
+% kubectl create -f redis-slave-service.yaml
+```
+
+You can verify that everything is running:
+
+```
+% kubectl get pods
+NAME                           READY     STATUS    RESTARTS   AGE
+redis-master-343230949-lpw91   1/1       Running   0          1d
+redis-slave-132015689-dpp46    1/1       Running   0          1d
+redis-slave-132015689-v06md    1/1       Running   0          1d
+```
+
+### Connecting the Guestbook frontend to Redis
+
+We're now going to use [Telepresence](http://www.telepresence.io) to create a virtual network between your local machine and the remote Kubernetes cluster. This way, the PHP application will be able to talk to remote cloud resources, and vice versa.
+
+We'll start by deploying the Telepresence proxy onto the Kubernetes cluster using the [`telepresence-deployment.yaml`](https://github.com/datawire/telepresence/blob/master/examples/guestbook/telepresence-deployment.yaml) deployment file.
+
+```
+% kubectl create -f telepresence-deployment.yaml
+```
+
+Next, we need to deploy an externally visible load balancer. The [`frontend-service.yaml`](https://github.com/datawire/telepresence/blob/master/examples/guestbook/frontend-service.yaml) uses a selector for `app:guestbook` and `tier:backend`. We've cleverly used these labels in the `telepresence-deployment.yaml`, so the load balancer will talk directly to our proxy.
+
+```
+% kubectl create -f frontend-service.yaml
+```
+
+Now, we're going to start the local Telepresence client, which will spawn a special shell which connects to the proxy in the remote Kubernetes cluster.
+
+```
+% telepresence --deployment telepresence-deployment --expose 8080 --run-shell
+```
+
+In this special shell, change to the `examples/guestbook` directory, and start the frontend application as follows. We'll need to know the directory where Predis is installed. You can figure this out by typing:
+
+```
+% pear config-get php_dir
+```
+
+Now, in the `examples/guestbook` directory, start PHP:
+
+```
+% php -d include_path="PATH_TO_PEAR_DIR" -S 0.0.0.0:8080
```

Comment:
It's PEAR, but I can see how this is confusing. I clarified the text above so explain.